src/proto/stream.rs

```diff
@@ -285,6 +288,18 @@ where
         self.pending_bytes += item.header_len() + item.payload.len();
         self.frame_queue.push_back(item);
     }
+
+    /// Sets the waker that is currently flushing to a new one and does nothing
+    /// if the waker is the same.
+    fn set_flushing_waker(&mut self, waker: &Waker) {
+        if !self
+            .flushing_waker
+            .as_ref()
+            .is_some_and(|waker| waker.will_wake(waker))
```

Comment:
You shadow the original waker here.
```suggestion
            .is_some_and(|w| w.will_wake(waker))
```